src/main/kotlin/com/faforever/userservice/security/FafJwtAuthenticationConverter.kt

```diff
@@ -13,15 +13,20 @@ data class FafAuthority(val role: String) : GrantedAuthority {
     override fun getAuthority() = "ROLE_$role"
 }
 
+data class FafScope(val role: String) : GrantedAuthority {
+    override fun getAuthority() = "SCOPE_$role"
+}
+
 /**
- * Extract the FAF roles that we put into the ext->roles array of the access token.
+ * Extract the FAF roles that we put into the ext->roles array of the access token and the scopes.
  */
 @Component
 class FafJwtAuthenticationConverter : Converter<Jwt, Flux<GrantedAuthority>> {
     override fun convert(jwt: Jwt): Flux<GrantedAuthority> {
         val ext: JSONObject? = jwt.claims["ext"] as? JSONObject
+        val scopes: JSONArray = jwt.claims["scp"] as JSONArray
```

Comment:
you don't need the variable type declarations if you use `as ...`

Reply:
what is this magic?